proto/p4/p4runtime.proto

```diff
@@ -136,6 +136,14 @@ message TableEntry {
   // this is stored, as long as it is returned with the rest of the entry in
   // a Read RPC.
   uint64 controller_metadata = 5;
+  // meter_config and counter_data are convenience fields that enable the
+  // controller to configure the direct resources associated with the table at
+  // the same time as a match-action entry is inserted or modified. When
+  // updating a table entry, leaving these fields unset means that the direct
+  // resources (if any) will not be updated. When reading a table entry, the
+  // values associated with the direct resources (if any) will be returned.
```

Comment:
I agree, we should revise the "Write/Read symmetry" section to clarify this. I think the essence is (as I understand) that mutations done by a Write should be visible on a subsequent Read. That Read may additionally return additional data, which should be fine. So its not strict bit-by-bit symmetry but more like logical symmetry.